### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by Behnam1369**
Add models based on the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+
```
Comment (mikethreels) [inline app/models/user.rb]: - [x] for the has_many relationship the foreign key should be set.
- [x] a user has_many likes as well.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  def last_3_posts
+    Post.where(author: self).last(3)
+  end
```
Comment (mikethreels) [inline app/models/user.rb]: - [x] this should be `posts` like in the has many relationship you wrote.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      author.posts_counter = 1
+    else
+      author.posts_counter += 1
+    end
+    author.save
+  end
+
+  def last_5_comments
+    Comment.where(post: self).last(5)
+  end
```
Comment (mikethreels) [inline app/models/post.rb]: - [x] this should be comments like in the has_many you created above

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Behnam, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Meltrust — APPROVED** [review]
Hi @Behnam1369,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To highlight

- Thank you for correcting your associations, and your methods as requested by the last reviewer✔️ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**